### DISCUSSION TITLE
[CONTRIBUTION] Speech Dataset Generator

### DISCUSSION THREAD

**Discussion by davidmartinrius** [speechbrain/speechbrain#2428] [Show and tell]
Hi everyone! 

I have just published this project on GitHub: https://github.com/davidmartinrius/speech-dataset-generator/

Now you can create datasets automatically with any audio or lists of audios.

I hope you can find it useful. 

Here are the key functionalities of the project:

1. **Dataset Generation:** The project allows for the creation of datasets with Mean Opinion Score (MOS).

2. **Silence Removal:** It includes a feature to remove silences from audio files, enhancing the overall quality.

3. **Sound Quality Improvement:** The project focuses on improving the quality of the audio.

4. **Audio Segmentation:** It can segment audio files within specified second ranges.

5. **Transcription:** The project transcribes the segmented audio, providing a textual representation.

6. **Gender Identification:** It identifies the gender of each speaker in the audio.

7. **Pyannote Embeddings:** Utilizes pyannote embeddings for speaker detection across multiple audio files.

8. **Automatic Speaker Naming:** Automatically assigns names to speakers detected in multiple audios.

9. **Multiple Speaker Detection:** Capable of detecting multiple speakers within each audio file.

Feel free to explore the project at https://github.com/davidmartinrius/speech-dataset-generator

**Comment by platform-kit** [comment]
Hey @davidmartinrius thank you for creating this tool, it was very much needed.

Would you be willing to add some config options for output format? 

For example, to train the [Metavoice model](https://github.com/metavoiceio/metavoice-src) you have to have a dataset in this format:
https://github.com/metavoiceio/metavoice-src/blob/main/datasets/sample_dataset.csv 

(One text file per audio sample.)

**Comment by davidmartinrius** [comment]
Absolutely @platform-kit , I think it will be quite easy to implement. I added a new task in https://github.com/davidmartinrius/speech-dataset-generator?tab=readme-ov-file#support-multiple-datasets

Keep in mind that I am programming this in my free time, so it may take a while. When implemented I will let you know in this thread. 

Thanks!

**Reply by platform-kit** [reply]
Amazing! thank you!

**Reply by davidmartinrius** [reply]
Hi @platform-kit I implemented the metavoice dataset inside the speech-dataset-generator.
You can install the latest version of speech-dataset-generator and then you can execute this inside the project folder. 

`python speech_dataset_generator/main.py --youtube_download https://www.youtube.com/watch\?v\=NW62DpzJ274 --output_directory ./output_directory --range_times 5-15 --datasets metavoice --enhancers deepfilternet resembleai`

This example uses a youtube video as an example, but you can use whatever you want as input, you can read the instructions to learn [how to do it](https://github.com/davidmartinrius/speech-dataset-generator?tab=readme-ov-file#examples).

Please, could you give it a star and share this project to your people? Thanks!

**Reply by platform-kit** [reply]
@davidmartinrius very impressive! thank you! Just starred and forked. 

Great work.
